src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -407,7 +407,7 @@ def launchCluster(self,
         createdSGs = self._createSecurityGroups()
         bdms = self._getBoto3BlockDeviceMappings(leader_type, rootVolSize=leaderStorage)
 
-        userData = self._getIgnitionUserData('leader')
+        userData = self._getIgnitionUserData('leader', self._architecture)
```

Comment:
This is going to pass the architecture as the second argument (`keyPath`). I think you need to say `architecture=self._architecture` here.